File: Java/Leccion6/Herencia/src/domain/Empleado.java

package domain;

public class Empleado extends Persona{
    private final int idEmpleado;
    private double sueldo;
    private static int contadorEmpleados;
    
    public Empleado(){ //constructor 1
        this.idEmpleado = ++Empleado.contadorEmpleados;
    }
    public Empleado(String nombre, double sueldo) { //constructor 2
//        super(nombre);
        this(); // Estamos llamando desde aquí al constructor vacío
        this.nombre = nombre; //podemos acceder directamente a los atributos
        //de la clase padre porque tienen el modificador de acceso protected
        this.sueldo = sueldo;
    }

    public int getIdEmpleado() {
        return idEmpleado;
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    @Override
    public String toString() { /*usamos clase StringBuilder*/
        StringBuilder sb = new StringBuilder();
        sb.append("Empleado{idEmpleado=").append(idEmpleado);
        sb.append(", sueldo=").append(sueldo);
        sb.append(", ").append(super.toString());
        sb.append('}');
        return sb.toString();
    }
    
    
    
    
    
    
    
}

File: Javascript3/Leccion7/Empleado.js
const { Persona } = require("./Persona.js"); // sintaxis modulos common js

class Empleado extends Persona{
    
    static contadorEmpleados = 0;

    constructor(nombre, apellido, edad, sueldo){
        super(nombre, apellido, edad);
        this._idEmpleado = ++Empleado.contadorEmpleados;
        this._sueldo = sueldo;
    }

    get idEmpleado(){
        return this._idEmpleado;
    }

    get sueldo(){
        return this._sueldo;
    }

    set sueldo(sueldo){
        this._sueldo = sueldo;
    }

    toString(){
        return super.toString()+" "+this._idEmpleado+" "+this._sueldo;
    }

}

module.exports = {
    Empleado : Empleado
}